airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -107,10 +107,13 @@
 
 
 class ModelToComponentFactory:
-    def __init__(self, limit_pages_fetched_per_slice: int = None, limit_slices_fetched: int = None):
+    def __init__(
+        self, limit_pages_fetched_per_slice: int = None, limit_slices_fetched: int = None, emit_connector_builder_messages: bool = False
```

Comment:
We had a recent discussion about also reducing the backoff rate when making subsequent requests for retryable errors. And that is really hinting that we should just have a general flag to pass to the factory to indicate that we should build components for the connector builder and cut down on the amount of args we pass in. However, I think we can address renaming this when that second use case is worked on.

It's not the best to have a connector builder specific context flag bleed into the factory implementation, but its also the lightest weight way to easily apply the decorator components across the whole factory.

Reply:
I'm not super opinionated on this. More flags makes it more difficult to use, but it allows us to better control the blast radius of features we implement for the connector builder.

Either way, I agree with waiting until we add a second flag to rename.